packages/odata-service-writer/src/data/defaults.ts

```diff
@@ -15,48 +18,108 @@ function setDefaultServicePath(service: OdataService): void {
 
 /**
  * Sets the default name for a given service.
- * If the service name is not defined, it sets the name to `DEFAULT_DATASOURCE_NAME`.
+ * Default serivce name is used only for first service.
  *
+ * @param {string} basePath - the root path of an existing UI5 application
  * @param {OdataService} service - The service object whose name needs to be set or modified.
+ * @param fs - the memfs editor instance
  */
-function setDefaultServiceName(service: OdataService): void {
-    service.name = service.name ?? DEFAULT_DATASOURCE_NAME;
+function setDefaultServiceName(basePath: string, service: OdataService, fs: Editor): void {
+    const manifestPath = join(basePath, 'webapp', 'manifest.json');
```

Comment:
Not sure if there is a better way to find the `manifest.json`, instead of hardcoded `webapp` - @heimwege?